pkg/blueprint/blueprints/pitr-postgres-blueprint.yaml

```diff
@@ -0,0 +1,345 @@
+        keyValue:
+          prefix: 'postgres-backups/{{ .StatefulSet.Name }}'
+          path: 'postgres-backups/{{ .StatefulSet.Name }}/{{ toDate "2006-01-02T15:04:05.999999999Z07:00" .Time  | date "2006-01-02T15-04-05" }}/manifest.txt'
+    secretNames:
+    - postgresql
+    phases:
+    - func: KubeExec
+      name: baseBackup
+      args:
+        namespace: "{{ .StatefulSet.Namespace }}"
+        pod: "{{ index .StatefulSet.Pods 0 }}"
+        container: "{{ .StatefulSet.Name }}"
+        command:
+          - bash
+          - -o
+          - errexit
+          - -o
+          - pipefail
+          - -o
+          - nounset
+          - -o
+          - xtrace
+          - -c
+          - |
+            env_dir="${PGDATA}/env"
+            mkdir -p "${env_dir}"
+            env_wal_prefix="${env_dir}/WALE_S3_PREFIX"
+
+            # We check for an existing timeline. If one does not exist, we create it based on the current time.
+            if [[ ! -e "${env_wal_prefix}" ]]
+            then
+              # Setup wal-e s3 connection parameters.
+              timeline={{ toDate "2006-01-02T15:04:05.999999999Z07:00" .Time  | date "2006-01-02T15-04-05" }}
+              wale_s3_prefix="s3://{{ .Profile.Location.Bucket }}/postgres-backups/{{ .StatefulSet.Name }}/${timeline}"
+              echo "${wale_s3_prefix}" > "${env_wal_prefix}"
+            fi
+
+            # Setup the other S3 parameters for wal-e
+            env_wal_endpoint="${env_dir}/WALE_S3_ENDPOINT"
+            env_wal_region="${env_dir}/AWS_REGION"
+            env_wal_access_key_id="${env_dir}/AWS_ACCESS_KEY_ID"
+            env_wal_secret_access_key="${env_dir}/AWS_SECRET_ACCESS_KEY"
+
+            {{- if .Profile.Location.Endpoint }}
+              wale_s3_endpoint="{{ .Profile.Location.Endpoint | quote }}"
+              wale_s3_endpoint=${wale_s3_endpoint,,}
+              {{- if .Profile.SkipSSLVerify }}
+                # Since wal-e does not support skip-ssl-verify switch to http://
+                wale_s3_endpoint="${wale_s3_endpoint/https/http}"
+              {{- end }}
+
+              #add the scheme that wal-e requires
+              wale_s3_endpoint="${wale_s3_endpoint//https\:\/\//https+path://}"
+              wale_s3_endpoint="${wale_s3_endpoint//http\:\/\//http+path://}"
+
+              echo "${wale_s3_endpoint}" > "${env_wal_endpoint}"
+
+              # Region will be ignored for S3 compatible object store so skipping.
+              rm -rf ${env_wal_region}
+            {{- else }}
+              # Region is required when no endpoint is used (AWS S3).
+              wale_s3_region="us-east-1"
+              {{- if .Profile.Location.Region }}
+                wale_s3_region="{{ .Profile.Location.Region | quote}}"
+              {{- end }}
+              echo "${wale_s3_region}" > "${env_wal_region}"
+            {{- end }}
+
+            set +o xtrace
+            echo "{{ .Profile.Credential.KeyPair.Secret }}" > "${env_wal_secret_access_key}"
```

Comment:
@PrasadG193 we made a few changes to pg blueprint, please take a look at this https://github.com/kanisterio/kanister/blob/master/examples/helm/kanister/kanister-postgresql/kanister/postgres-blueprint.yaml#L73

Reply:
@SupriyaKasten I have addressed your comments and opened seperate PR https://github.com/kanisterio/kanister/pull/428